funsor/tensor.py

```diff
@@ -326,7 +326,7 @@ def unscaled_sample(self, sampled_vars, sample_inputs):
         #   Then g is an unbiased estimator of f in value and all derivatives.
         #   In the special case f = detach(f), we can simplify to
         #       g = delta(x=x0) |f|.
-        if flat_logits.requires_grad:
+        if hasattr(flat_logits, "requires_grad") and flat_logits.requires_grad:
```

Comment:
I am not sure if for numpy backend, we should exercise `if` or `else` branch here.